model/validation.go

```diff
@@ -204,16 +202,433 @@ func (t Tags) Validate() error {
+	parts := strings.Split(subnet, "/")
+	if len(parts) > 2 {
+		return fmt.Errorf("%q is not valid CIDR notation", subnet)
+	}
+
+	var errs error
+
+	if len(parts) == 2 {
+		if err := validateCIDRBlock(parts[1]); err != nil {
+			errs = multierror.Append(errs, err)
+		}
+	}
+
+	if err := validateIPv4Address(parts[0]); err != nil {
+		errs = multierror.Append(errs, err)
+	}
+
+	return errs
+}
+
+// validateCIDRBlock validates that a string in "CIDR notation" or "Dot-decimal notation"
+func validateCIDRBlock(cidr string) error {
+	if bits, err := strconv.Atoi(cidr); err != nil || bits <= 0 || bits > 32 {
+		return fmt.Errorf("/%v is not a valid CIDR block", cidr)
+	}
+
+	return nil
+}
+
+// validateIPv4Address validates that a string in "CIDR notation" or "Dot-decimal notation"
+func validateIPv4Address(addr string) error {
+	octets := strings.Split(addr, ".")
+	if len(octets) != 4 {
+		return fmt.Errorf("%q is not a valid IP address", addr)
+	}
+
+	for _, octet := range octets {
+		if n, err := strconv.Atoi(octet); err != nil || n < 0 || n > 255 {
+			return fmt.Errorf("%q is not a valid IP address", addr)
+		}
+	}
+
+	return nil
+}
+
+func validateDelay(delay *proxyconfig.HTTPFaultInjection_Delay) (errs error) {
+
+	errs = validateFloatPercent(errs, delay.Percent, "delay")
+
+	if delay.GetFixedDelaySeconds() < 0 {
+		errs = multierror.Append(errs, fmt.Errorf("delay fixed_seconds invalid"))
+	}
+
+	if delay.GetExponentialDelaySeconds() != 0 {
+		if delay.GetExponentialDelaySeconds() < 0 {
+			errs = multierror.Append(errs, fmt.Errorf("delay exponential_seconds invalid"))
+		}
+		errs = multierror.Append(errs, fmt.Errorf("Istio does not support exponential_seconds yet"))
+	}
+
+	return
+}
+
+func validateAbortHTTPStatus(httpStatus *proxyconfig.HTTPFaultInjection_Abort_HttpStatus) (errs error) {
+
+	if httpStatus.HttpStatus < 0 || httpStatus.HttpStatus > 600 {
+		errs = multierror.Append(errs, fmt.Errorf("invalid abort http status %v", httpStatus.HttpStatus))
+	}
+
+	return
+}
+
+func validateAbort(abort *proxyconfig.HTTPFaultInjection_Abort) (errs error) {
+
+	errs = validateFloatPercent(errs, abort.Percent, "abort")
+
+	switch abort.ErrorType.(type) {
+	case *proxyconfig.HTTPFaultInjection_Abort_GrpcStatus:
+		// TODO No validation yet for grpc_status / http2_error / http_status
+		errs = multierror.Append(errs, fmt.Errorf("Istio does not support gRPC fault injection yet"))
+	case *proxyconfig.HTTPFaultInjection_Abort_Http2Error:
+		// TODO No validation yet for grpc_status / http2_error / http_status
+	case *proxyconfig.HTTPFaultInjection_Abort_HttpStatus:
+		if err := validateAbortHTTPStatus(abort.ErrorType.(*proxyconfig.HTTPFaultInjection_Abort_HttpStatus)); err != nil {
+			errs = multierror.Append(errs, err)
+		}
+	}
+
+	// No validation yet for override_header_name
+
+	return
+}
+
+func validateTerminate(terminate *proxyconfig.L4FaultInjection_Terminate) (errs error) {
+
+	errs = validateFloatPercent(errs, terminate.Percent, "terminate")
```

Comment:
nit: could be a single line: `return validateFloatPercent(errs, terminate.Percent, "terminate")`